### PR TITLE
remove note about old node for Apple Silicon

### PR DESCRIPTION
**Pull request by cat5inthecradle**
Tested and confirmed that `nvm use 18` runs without issue on M1.

Other Apple Silicon users should confirm. Tagging some known M1 users.

## Links

Follow up from #51533

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by sureshc — APPROVED** [review]
Thanks!

**Review by snickell — APPROVED** [review]
LGTM, and can confirm I'm using node w/o rosetta after the Node 18 upgrade. Dropping this block of text makes sense to me.

That said, if you're following `SETUP.md` as written, this section was technically redundant... because you're instructed in MacOS Step 1 to run all these commands in a rosetta/x86 terminal.

A larger SETUP.md rework is required to remove all rosetta refs, see https://github.com/code-dot-org/code-dot-org/pull/51431 which is largely SETUP.md changes after splitting out node and selenium upgrades. 

However, we'll need an upgrade to the `webdriver` gem before we can fully say that you don't need rosetta: as it stands UI tests won't work on Apple Silicon. #51635 does a lot of the selenium upgrade (the step from selenium 3 to 4 is bigger than 4 to 5), but I believe an even newer version is required for M1 compat.

TL;DR: node 18 works native on apple silicon now, but *if* you care about running UI tests, you should still be running ruby / `bundle exec` commands etc in Rosetta. If you don't care about UI tests, you can skip rosetta entirely.
